src/BloomBrowserUI/bookEdit/OverflowChecker/OverflowChecker.ts

```diff
@@ -0,0 +1,217 @@
+/// <reference path="../../lib/jquery.d.ts" />
+/// <reference path="../StyleEditor/StyleEditor.ts" />
+
+interface qtipInterface extends JQuery {
+    qtip(options: string): JQuery;
+}
+
+class OverflowChecker {
+
+    // When a div is overfull, these handlers will add the overflow class so it gets a red background or something
+    // But this function should just do some basic checks and ADD the HANDLERS!
+    public AddOverflowHandlers(container:HTMLElement) {
+        //NB: for some historical reason in March 2014 the calendar still uses textareas
+        var queryElementsThatCanOverflow = ".bloom-editable, textarea";
+        var editablePageElements = $(container).find(queryElementsThatCanOverflow);
+
+        //first, check to see if the stylesheet is going to give us overflow even for a single character:
+        editablePageElements.each(function () {
+            OverflowChecker.CheckOnMinHeight(this);
+        });
+
+        //Add the handler so that when the elements change, we test for overflow
+        editablePageElements.on("keyup paste", function (e) {
+            var target = e.target;
+            // Give the browser time to get the pasted text into the DOM first, before testing for overflow
+            // GJM -- One place I read suggested that 0ms would work, it just needs to delay one 'cycle'.
+            //        At first I was concerned that this might slow typing, but it doesn't seem to.
+            setTimeout(function () {
+                OverflowChecker.MarkOverflowInternal(target);
+
+                //REVIEW: why is this here, in the overflow detection?
+
+                // This will make sure that any language tags on this div stay in position with editing.
+                // Reposition all language tips, not just the tip for this item because sometimes the edit moves other controls.
+                (<qtipInterface>$(queryElementsThatCanOverflow)).qtip('reposition');
+            }, 100); // 100 milliseconds
+            e.stopPropagation();
+        });
+
+        // Add another handler so that when the user resizes an origami pane, we check the overflow again
+        $(container).find(".split-pane-component-inner").bind('_splitpaneparentresize', function () {
+            var $this = $(this);
+            $this.find(queryElementsThatCanOverflow).each(function () {
+                OverflowChecker.MarkOverflowInternal(this);
+            });
+        });
+
+        // Right now, test to see if any are already overflowing
+        editablePageElements.each(function () {
+            OverflowChecker.MarkOverflowInternal(this);
+        });
+    }
+
+    // Actual testable determination of Type I overflow or not
```

Comment:
If you explained somewhere what Type 1 and 2 overflow are, I missed it. An explanation or cross-ref would be helpful.